File: Backend/src/main/java/com/SEPT_Backend/Backend/Service/BookingService.java
package com.SEPT_Backend.Backend.Service;

import java.util.List;  

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.SEPT_Backend.Backend.model.Booking;
import com.SEPT_Backend.Backend.repository.BookingRepository;

/*
 * Booking operations are called from here into 
 * the repository 
 */
@Service
public class BookingService {
	
	@Autowired
	private BookingRepository  bookingRepo ;

	public Booking saveBooking(Booking booking) {
		return bookingRepo.save(booking);
	}
	
	//get all bookings
	public List<Booking> getBookings(){
		return bookingRepo.findAll();
	}

	//find if booking already exists
	public List<Booking> findByDayAndTime(String day, String time, String name) {
		return bookingRepo.findByDayAndTime(day, time, name);
	}

	//get list of booking using username
	public List<Booking> findBooking(String username) {
		return bookingRepo.findBooking(username);
	}
	
	//find a booking by id
	public Booking findByID(long id) {
		return bookingRepo.findByID(id);
	}
	
	//delete a booking
	public void deleteBooking(long id){
		bookingRepo.deleteBooking(id);
	}

	//change the status to 'approved'
	public void approveBooking(long id) {
		bookingRepo.approveBooking(id);
	}
	
	//change the status to 'unapproved'
	public void rejectBooking(long id) {	
		bookingRepo.rejectBooking(id);	
	}
	
	//delete the booking from the database
	public void deleteBooking(Booking booking){
		bookingRepo.delete(booking);
	}
	
}

File: FrontEnd/src/components/services/BookingService.js
import axios from 'axios';

import authHeader from './AuthHeader';

const BOOKING_API_BASE_URL = "http://localhost:8080/api/v1/booking";


class BookingService {

    //add a booking of a customer
    addBooking(booking) {
        return axios.post(BOOKING_API_BASE_URL + "/add", booking, { headers: authHeader() });
    }

    //find a booking of a customer using username
    findBooking(username) {
        return axios.post(BOOKING_API_BASE_URL + "/info", username, { headers: authHeader() });
    }

    //get all the bookings
    getBooking() {
        return axios.get(BOOKING_API_BASE_URL + "/all", { headers: authHeader() });
    }

    //Approve an booking by changing the corresponding ID in the DB
    approveBooking(id) {
        return axios.post(BOOKING_API_BASE_URL + "/approve", id, { headers: authHeader() });
    }

    //Reject the booking in the DB 
    rejectBooking(id) {
        return axios.post(BOOKING_API_BASE_URL + "/reject", id, { headers: authHeader() });
    }

    //Remove a booking from the DB
    deleteBooking(id) {
        return axios.post(BOOKING_API_BASE_URL + "/delete", id, { headers: authHeader() });
    }

}

export default new BookingService()